New envSlice function

potatop opened the pull request:
I added envSlice function so you can do range templates. For example
`ENV_TENANTS=bam:ham:pam`
and template 
```
{{range envSlice "ENV_TENANTS"}}
  path: {{.}}/somepath
{{end}}
```
would produce
```
  path: bam/somepath
  path: ham/somepath
  path: pam/somepath
```

9len commented on envtemplate.go:
```diff
@@ -196,6 +203,14 @@ func (r *runner) envOrDefault(key, defValue string) string {
 	return value
 }
 
+func (r *runner) envSlice(key string) ([]string, error) {
```
let's call it `envSplit` instead, and make it take the delimiter as an explicit argument.

9len reviewed (commented):
Thanks for the contribution! After you address the feedback, can you rebase your changes as a single commit? Then I will merge it to our internal monorepo and push it back out.

9len commented on envtemplate.go:
```diff
@@ -46,6 +47,10 @@ Two functions are made avaiable to the templates:
     {{print "{{envOrDefault \"TBN_HOME\" \"~/$TBN_WORKSPACE/tbn\"}}"}}
 
+{{ul "envSplit"}}: used to specify a required environment variable
+whose value is a list separated by ":" character:
```
doc needs an update still.

Commit:
add envSplit function.

envSplit function slices an environment variable with multiple values
separated by sep and returns a slice of all the substrings between those
separators.

9len reviewed (commented):
looks great! will get it merged down on our side. thanks!

potatop commented:
Thanks. Glads you find this potentially useful.
